GCC 6.1.1 build error.

Distribution: Arch Linux (testing repo activate)

local/gcc 6.1.1-1 (base-devel)

CPPFLAGS="-D_FORTIFY_SOURCE=2"
CFLAGS="-march=native -O2 -pipe -fstack-protector-strong"
CXXFLAGS="-march=native -O2 -pipe -fstack-protector-strong"
LDFLAGS="-Wl,-O1,--sort-common,--as-needed,-z,relro"
MAKEFLAGS="-j1" 
DEBUG_CFLAGS="-g -fvar-tracking-assignments"
DEBUG_CXXFLAGS="-g -fvar-tracking-assignments"

 gcc -march=native  -Q --help=target -fverbose-asm

`The following options are target specific:
  -m128bit-long-double                  [disabled]
  -m16                                  [disabled]
  -m32                                  [disabled]
  -m3dnow                               [disabled]
  -m3dnowa                              [disabled]
  -m64                                  [enabled]
  -m80387                               [enabled]
  -m8bit-idiv                           [disabled]
  -m96bit-long-double                   [enabled]
  -mabi=                                sysv
  -mabm                                 [disabled]
  -maccumulate-outgoing-args            [disabled]
  -maddress-mode=                       short
  -madx                                 [disabled]
  -maes                                 [disabled]
  -malign-data=                         compat
  -malign-double                        [disabled]
  -malign-functions=                    0
  -malign-jumps=                        0
  -malign-loops=                        0
  -malign-stringops                     [enabled]
  -mandroid                             [disabled]
  -march=                               ivybridge
  -masm=                                att
  -mavx                                 [enabled]
  -mavx2                                [disabled]
  -mavx256-split-unaligned-load         [disabled]
  -mavx256-split-unaligned-store        [disabled]
  -mavx512bw                            [disabled]
  -mavx512cd                            [disabled]
  -mavx512dq                            [disabled]
  -mavx512er                            [disabled]
  -mavx512f                             [disabled]
  -mavx512ifma                          [disabled]
  -mavx512pf                            [disabled]
  -mavx512vbmi                          [disabled]
  -mavx512vl                            [disabled]
  -mbionic                              [disabled]
  -mbmi                                 [disabled]
  -mbmi2                                [disabled]
  -mbranch-cost=                        0
  -mcld                                 [disabled]
  -mclflushopt                          [disabled]
  -mclwb                                [disabled]
  -mclzero                              [disabled]
  -mcmodel=                             32
  -mcpu=                      
  -mcrc32                               [disabled]
  -mcx16                                [enabled]
  -mdispatch-scheduler                  [disabled]
  -mdump-tune-features                  [disabled]
  -mf16c                                [enabled]
  -mfancy-math-387                      [enabled]
  -mfentry                              [enabled]
  -mfma                                 [disabled]
  -mfma4                                [disabled]
  -mforce-drap                          [disabled]
  -mfp-ret-in-387                       [enabled]
  -mfpmath=                             387
  -mfsgsbase                            [enabled]
  -mfused-madd                
  -mfxsr                                [enabled]
  -mglibc                               [enabled]
  -mhard-float                          [enabled]
  -mhle                                 [disabled]
  -miamcu                               [disabled]
  -mieee-fp                             [enabled]
  -mincoming-stack-boundary=            0
  -minline-all-stringops                [disabled]
  -minline-stringops-dynamically        [disabled]
  -mintel-syntax              
  -mlarge-data-threshold=               0x10000
  -mlong-double-128                     [disabled]
  -mlong-double-64                      [disabled]
  -mlong-double-80                      [enabled]
  -mlwp                                 [disabled]
  -mlzcnt                               [disabled]
  -mmemcpy-strategy=          
  -mmemset-strategy=          
  -mmitigate-rop                        [disabled]
  -mmmx                                 [enabled]
  -mmovbe                               [disabled]
  -mmpx                                 [disabled]
  -mms-bitfields                        [disabled]
  -mmusl                                [disabled]
  -mmwaitx                              [disabled]
  -mno-align-stringops                  [disabled]
  -mno-default                          [disabled]
  -mno-fancy-math-387                   [disabled]
  -mno-push-args                        [disabled]
  -mno-red-zone                         [disabled]
  -mno-sse4                             [disabled]
  -mnop-mcount                          [disabled]
  -momit-leaf-frame-pointer             [disabled]
  -mpc32                                [disabled]
  -mpc64                                [disabled]
  -mpc80                                [disabled]
  -mpclmul                              [enabled]
  -mpcommit                             [disabled]
  -mpku                                 [disabled]
  -mpopcnt                              [enabled]
  -mprefer-avx128                       [disabled]
  -mpreferred-stack-boundary=           0
  -mprefetchwt1                         [disabled]
  -mprfchw                              [disabled]
  -mpush-args                           [enabled]
  -mrdrnd                               [disabled]
  -mrdseed                              [disabled]
  -mrecip                               [disabled]
  -mrecip=                    
  -mrecord-mcount                       [disabled]
  -mred-zone                            [enabled]
  -mregparm=                            0
  -mrtd                                 [disabled]
  -mrtm                                 [disabled]
  -msahf                                [enabled]
  -msha                                 [disabled]
  -mskip-rax-setup                      [disabled]
  -msoft-float                          [disabled]
  -msse                                 [enabled]
  -msse2                                [enabled]
  -msse2avx                             [disabled]
  -msse3                                [enabled]
  -msse4                                [enabled]
  -msse4.1                              [enabled]
  -msse4.2                              [enabled]
  -msse4a                               [disabled]
  -msse5                      
  -msseregparm                          [disabled]
  -mssse3                               [enabled]
  -mstack-arg-probe                     [disabled]
  -mstack-protector-guard=              tls
  -mstackrealign                        [enabled]
  -mstringop-strategy=                  [default]
  -mstv                                 [disabled]
  -mtbm                                 [disabled]
  -mtls-dialect=                        gnu
  -mtls-direct-seg-refs                 [enabled]
  -mtune-ctrl=                
  -mtune=                               ivybridge
  -muclibc                              [disabled]
  -mveclibabi=                          [default]
  -mvect8-ret-in-mem                    [disabled]
  -mvzeroupper                          [disabled]
  -mx32                                 [disabled]
  -mxop                                 [disabled]
  -mxsave                               [enabled]
  -mxsavec                              [disabled]
  -mxsaveopt                            [enabled]
  -mxsaves                              [disabled]

  Known assembler dialects (for use with the -masm-dialect= option):
    att intel

  Known ABIs (for use with the -mabi= option):
    ms sysv

  Known code models (for use with the -mcmodel= option):
    32 kernel large medium small

  Valid arguments to -mfpmath=:
    387 387+sse 387,sse both sse sse+387 sse,387

  Known data alignment choices (for use with the -malign-data= option):
    abi cacheline compat

  Known vectorization library ABIs (for use with the -mveclibabi= option):
    acml svml

  Known address mode (for use with the -maddress-mode= option):
    long short

  Known stack protector guard (for use with the -mstack-protector-guard= option):
    global tls

  Valid arguments to -mstringop-strategy=:
    byte_loop libcall loop rep_4byte rep_8byte rep_byte unrolled_loop vector_loop

  Known TLS dialects (for use with the -mtls-dialect= option):
    gnu gnu2
`

`[ 90%] Building CXX object CMakeFiles/qtermwidget5.dir/lib/moc_ScreenWindow.cpp.o
[ 92%] Building CXX object CMakeFiles/qtermwidget5.dir/lib/moc_SearchBar.cpp.o
In file included from /tmp/yaourt-tmp-narwhal/aur-qtermwidget-git/src/qtermwidget/lib/Vt102Emulation.cpp:24:0:
/tmp/yaourt-tmp-narwhal/aur-qtermwidget-git/src/qtermwidget/lib/Vt102Emulation.h:98:16: error: 'virtual void Konsole::Vt102Emulation::focusLost()' marked 'override', but does not override
   virtual void focusLost() Q_DECL_OVERRIDE;
                ^~~~~~~~~
In file included from /tmp/yaourt-tmp-narwhal/aur-qtermwidget-git/src/qtermwidget/lib/Vt102Emulation.cpp:24:0:
/tmp/yaourt-tmp-narwhal/aur-qtermwidget-git/src/qtermwidget/lib/Vt102Emulation.h:99:16: error: 'virtual void Konsole::Vt102Emulation::focusGained()' marked 'override', but does not override
   virtual void focusGained() Q_DECL_OVERRIDE;
                ^~~~~~~~~~~
CMakeFiles/qtermwidget5.dir/build.make:646: recipe for target 'CMakeFiles/qtermwidget5.dir/lib/Vt102Emulation.cpp.o' failed
make[2]: *** [CMakeFiles/qtermwidget5.dir/lib/Vt102Emulation.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /tmp/yaourt-tmp-narwhal/aur-qtermwidget-git/src/qtermwidget/lib/Session.cpp:46:0:
/tmp/yaourt-tmp-narwhal/aur-qtermwidget-git/src/qtermwidget/lib/Vt102Emulation.h:98:16: error: 'virtual void Konsole::Vt102Emulation::focusLost()' marked 'override', but does not override
   virtual void focusLost() Q_DECL_OVERRIDE;
                ^~~~~~~~~
In file included from /tmp/yaourt-tmp-narwhal/aur-qtermwidget-git/src/qtermwidget/lib/Session.cpp:46:0:
/tmp/yaourt-tmp-narwhal/aur-qtermwidget-git/src/qtermwidget/lib/Vt102Emulation.h:99:16: error: 'virtual void Konsole::Vt102Emulation::focusGained()' marked 'override', but does not override
   virtual void focusGained() Q_DECL_OVERRIDE;
                ^~~~~~~~~~~
CMakeFiles/qtermwidget5.dir/build.make:526: recipe for target 'CMakeFiles/qtermwidget5.dir/lib/Session.cpp.o' failed
make[2]: *** [CMakeFiles/qtermwidget5.dir/lib/Session.cpp.o] Error 1
CMakeFiles/Makefile2:99: recipe for target 'CMakeFiles/qtermwidget5.dir/all' failed
make[1]: *** [CMakeFiles/qtermwidget5.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
`

Also, last stable build of qtermwidget (0.6.0) build correctly.

--- a/lib/Vt102Emulation.h
+++ b/lib/Vt102Emulation.h
@@ -95,8 +95,8 @@ public slots:
   virtual void sendText(const QString& text);
   virtual void sendKeyEvent(QKeyEvent*);
   virtual void sendMouseEvent(int buttons, int column, int line, int eventType);
-  virtual void focusLost() Q_DECL_OVERRIDE;
-  virtual void focusGained() Q_DECL_OVERRIDE;
+  virtual void focusLost();
+  virtual void focusGained();
 
 protected:
   // reimplemented from Emulation

